modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ObjectRelationship/ObjectRelationship.js

```diff
@@ -12,44 +12,130 @@
  * details.
  */
 
-import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocomplete';
+import ClayAutocomplete from '@clayui/autocomplete';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import {useDebounce} from '@clayui/shared';
 import {FieldBase} from 'dynamic-data-mapping-form-field-type/FieldBase/ReactFieldBase.es';
+import {fetch} from 'frontend-js-web';
 import React from 'react';
 
-const ObjectRelationship = ({
+const NETWORK_STATUS_LOADING = 1;
+const NETWORK_STATUS_UNUSED = 4;
+
+const LoadingWithDebounce = ({loading, render}) => {
+	const debouncedLoadingChange = useDebounce(loading, 500);
+
+	if (loading || debouncedLoadingChange) {
+		return (
+			<ClayDropDown.Item className="disabled">
+				{Liferay.Language.get('loading')}
+			</ClayDropDown.Item>
+		);
+	}
+
+	return render;
+};
+
+export function ObjectRelationship({
 	apiURL,
-	initialLabel,
-	initialValue,
+	initialLabel = '',
+	initialValue = '',
 	inputName,
-	itemsKey,
-	itemsLabel,
+	labelKey = 'label',
 	name,
-	onChange,
 	readOnly,
 	required,
-	value,
+	valueKey = 'value',
 	...otherProps
-}) => (
-	<FieldBase
-		name={name}
-		readOnly={readOnly}
-		required={required}
-		{...otherProps}
-	>
-		<Autocomplete
-			apiUrl={apiURL}
-			initialLabel={initialLabel}
-			initialValue={initialValue}
-			inputName={inputName}
-			itemsKey={itemsKey}
-			itemsLabel={itemsLabel}
+}) {
+	const [selectedValue, setSelectedValue] = React.useState(initialValue);
+	const [inputValue, setInputValue] = React.useState(initialLabel);
+	const [networkStatus, setNetworkStatus] = React.useState(
+		NETWORK_STATUS_LOADING
+	);
+
+	const {resource} = useResource({
+		fetch,
+		fetchPolicy: 'cache-first',
+		link: apiURL,
+		onNetworkStatusChange: setNetworkStatus,
+		variables: {
+			page: 1,
+			pageSize: 10,
+			search: inputValue,
+		},
+	});
+
+	const loading = networkStatus < NETWORK_STATUS_UNUSED;
+
+	return (
+		<FieldBase
 			name={name}
-			onChange={onChange}
 			readOnly={readOnly}
 			required={required}
-			value={value}
-		/>
-	</FieldBase>
-);
+			{...otherProps}
+		>
+			<input
+				id={inputName}
+				name={inputName}
+				type="hidden"
+				value={selectedValue}
+			/>
```

Comment:
I think I don't even need this hidden input here because FieldBase will take care of that because it also handles various values depending on the language.